### PR TITLE
Remove readiness probe and increase initial delay

### LINKED ISSUE #1419: Probes fail for development deployment
- Issue type: Bug report

### Short description
Probes for development branch fail with `503` for new deployments because of DOS protection in Jetty.

### PR DESCRIPTION
**Pull request by singhpratyush**
### Short description

Fixes #1419.

I have:
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only strictly only one commit per issue.

### For the reviewers
I have:
- [ ] Reviewed this pull request by an authorized contributor.
- [ ] The reviewer is assigned to the pull request.

### CODE REVIEW

**Comment by singhpratyush** [comment]
@Achint08 @djmgit @hemantjadon @kavithaenair @SKrPl @vibhcool: Please review.

File: kubernetes/yamls/development/api-server/api-deployment.yml [hunk 1]
```diff
@@ -18,13 +18,8 @@ spec:
...
-          initialDelaySeconds: 30
-          timeoutSeconds: 3
-        readinessProbe:
-          httpGet:
-            path: /api/status.json
-            port: 80
-          initialDelaySeconds: 30
+          initialDelaySeconds: 90
+          periodSeconds: 300
           timeoutSeconds: 3
```
Comment (vibhcool) [inline kubernetes/yamls/development/api-server/api-deployment.yml]: could you try one thing, this DOS attack is due to high no. of requests in less time. 
add `periodSeconds: 5` in readinessprobe

Comment (vibhcool) [inline kubernetes/yamls/development/api-server/api-deployment.yml]: also try it for values more than 10

Comment (singhpratyush) [inline kubernetes/yamls/development/api-server/api-deployment.yml]: @vibhcool: I've tried with even larger values, but I still get the error. Can you please check if it works for your deployment.

Comment (vibhcool) [inline kubernetes/yamls/development/api-server/api-deployment.yml]: @singhpratyush this creates the deployment but doesn't work
```
vibhorverma291995@loklak-server-175311:~/loklak_server$ ./kubernetes/bin/deploy-development.sh create
Creating objects from configurations.
Make sure that persistent disk is already created.

Creating Elasticsearch Replication Controller
namespace "elasticsearch" created
deployment "es" created
service "elasticsearch" created
serviceaccount "elasticsearch" created
Waiting for server to start up. ~20s.
Creating loklak deployment
namespace "web" created
deployment "server" created
service "server" created
configmap "server" created
Trying to fetch public IP. ~40s.
NAME      CLUSTER-IP      EXTERNAL-IP   PORT(S)        AGE
server    10.55.248.240   <pending>     80:30575/TCP   41s
Deployed loklak on Kubernetes
vibhorverma291995@loklak-server-175311:~/loklak_server$ kubectl get services --namespace=web
NAME      CLUSTER-IP      EXTERNAL-IP      PORT(S)        AGE
server    10.55.248.240   35.197.230.229   80:30575/TCP   1m
vibhorverma291995@loklak-server-175311:~/loklak_server$
```

Comment (vibhcool) [inline kubernetes/yamls/development/api-server/api-deployment.yml]: the livelinessprobe still fails `Liveness:           http-get http://:80/api/status.json delay=90s timeout=3s period=300s #success=1 #failure=3`

Comment (vibhcool) [inline kubernetes/yamls/development/api-server/api-deployment.yml]: the livelinessprobe and outputs this as `http://:80/api/status.json`, shouldn't this be an error?

Comment (singhpratyush) [inline kubernetes/yamls/development/api-server/api-deployment.yml]: > `Liveness: http-get http://:80/api/status.json delay=90s timeout=3s period=300s #success=1 #failure=3`

This is not the failure message, it is just showing the configuration for probes.

You need to see the events for the pod in order to know if the probes are passing.

Comment (vibhcool) [inline kubernetes/yamls/development/api-server/api-deployment.yml]: Error logs: https://gist.github.com/vibhcool/e6188aa72cb11cc04d0286dfb5f84bfc

Comment (vibhcool) [inline kubernetes/yamls/development/api-server/api-deployment.yml]: I fixed this error by adding missing line here -> https://github.com/singhpratyush/loklak_server/compare/ee4ad16446d2f58fa0f47b537a8ddbcc82eb4c86...vibhcool:60a975a62fdf10ddca4a976da8a479620153df27

Comment (vibhcool) [inline kubernetes/yamls/development/api-server/api-deployment.yml]: Now the error I am getting is -> https://gist.github.com/vibhcool/57aaffd32707717b52745824305a37e7

Comment (singhpratyush) [inline kubernetes/yamls/development/api-server/api-deployment.yml]: I'm not sure about why you are getting this error as this never happened to me. Is the ES deployment running before you create the server deployment?

Comment (vibhcool) [inline kubernetes/yamls/development/api-server/api-deployment.yml]: nopes, before creating ES, there is no existing pod.

Comment (singhpratyush) [inline kubernetes/yamls/development/api-server/api-deployment.yml]: Well, I have tried with all possibilities and things are working fine. You can create the deployments one by one and check for their working - 
```bash
kubectl create -R -f kubernetes/yamls/development/elasticsearch
# Wait for elasticsearch to get ready
kubectl create -R -f kubernetes/yamls/development/api-server
```

Comment (vibhcool) [inline kubernetes/yamls/development/api-server/api-deployment.yml]: @singhpratyush , please go through this. This is the exact logs from creation of server to the error I got. Please see where I am going wrong. This time I didn't use deploy script, instead I wrote them as you suggested.
https://gist.github.com/vibhcool/58ed3f11ee504a6760c123d6beeba7f5

Server deployment: http://35.189.73.55/

Comment (singhpratyush) [inline kubernetes/yamls/development/api-server/api-deployment.yml]: * How do you know that Elasticsearch deployment is running fine? Can you please check the logs for Elasticsearch?
* Have you created the persistent disk?
* What is transport address pointing to in the configs?
    ```bash
     kubectl exec -it <pod> --namespace=web -c server bash
     # You are now in server pod
     cat loklak_server/conf/config.properties  # Look for transport address
    ```

Comment (vibhcool) [inline kubernetes/yamls/development/api-server/api-deployment.yml]: 1) I didn't say ElasticSearch is working fine. I showed you logs that says it isn't working fine :sweat_smile:
2) yes, persistant disk exists
3) Everything is as in the PR
```
# External elasticsearch:
# If elasticsearch_transport.enabled is set to true, then an external elasticsearch is used
# The external elasticsearch server must have one configuration change to enable the access of
# loklak: in elasticsearch/config/elasticsearch.yml must be added one line, containing
# cluster.name: loklakcluster
elasticsearch_transport.enabled = false
elasticsearch_transport.addresses = 127.0.0.1:9300
elasticsearch_transport.cluster.name = loklakcluster
elasticsearch_requiredClusterHealthStatus = YELLOW

# The following are elasticsearch properties:
# The prefix 'elasticseach.' is removed and handed over to the elasticsearch node
elasticsearch.cluster.name=index
elasticsearch.cluster.routing.allocation.disk.threshold_enabled=true
elasticsearch.cluster.routing.allocation.disk.watermark.low=4gb
elasticsearch.cluster.routing.allocation.disk.watermark.high=2gb
elasticsearch.http.enabled=true
elasticsearch.http.cors.enabled=true
elasticsearch.http.cors.allow-origin=/https?:\/\/localhost(:[0-9]+)?/
elasticsearch.http.port=9200
elasticsearch.transport.tcp.port=9300
elasticsearch.network.host=localhost
elasticsearch.node.data=true
elasticsearch.node.master=true
elasticsearch.index.number_of_shards=8
elasticsearch.index.number_of_replicas=0
elasticsearch.index.auto_expand_replicas=false
elasticsearch.index.refresh_interval=10s
elasticsearch.index.max_result_window=10000
elasticsearch.path.conf=conf
elasticsearch.path.data=data
elasticsearch.path.home=data
elasticsearch.elasticsearch_discovery_zen_ping_multicast_enabled = false
elasticsearch.indices.fielddata.cache.size=10%
elasticsearch.index.translog.flush_threshold_size=8mb
#elasticsearch.path.logs=data/log
```

Comment (singhpratyush) [inline kubernetes/yamls/development/api-server/api-deployment.yml]: So here is the issue - Elasticsearch is not present at `127.0.0.1`. It is there are `elasticsearch.elasticsearch:9300`. Please use proper image to set things up.

**Pull request merged**